prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -2491,4 +2447,69 @@ fun String.trimAndTruncate(maxLength: Int?): String {
         startTrimmed
     }
     return truncated.trimEnd()
+}
+
+/**
+ * Class for processing HL7 HAPI library exceptions that derive from HL7Exception
+ */
+class HL7HapiErrorProcessor : Logging {
```

Comment:
This is the "main" part of the PR and is how we now determine which error code we should apply to a HAPI exception. This is done by taking the "field" that is found in the HAPI exception message and then checking if the Schema object has an element with the same name and then seeing what type that element is associated with.